### ISSUE TITLE
semi-colon expected after first class in Tailwind @apply rules inside .vue file

### ISSUE THREAD

**Issue by geert-plessers** [stylelint/stylelint#5127]
> Clearly describe the bug

When applying multiple classes in a single Tailwind @apply rule **in a .vue file**. The error "semi-colon expected" gets shown after the first class.

> Which rule, if any, is the bug related to?

`scss(css-semicolonexpected)`

> What code is needed to reproduce the bug?

e.g.

```html
<style lang="scss" scoped>
.btn {
    @apply flex font-bold text-14 items-center justify-center w-full bg-blue text-white border border-blue focus:outline-none hover:bg-transparent hover:text-blue disabled:opacity-50 transition-colors;
}
</style>
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
    rules: {
        'at-rule-no-unknown': [
            true,
            {
                ignoreAtRules: ['tailwind', 'apply', 'responsive', 'variants', 'screen'],
            },
        ],
        'declaration-block-trailing-semicolon': null,
        'no-descending-specificity': null,
    },
}
```

> Which version of stylelint are you using?

`13.9.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

It's related to scss style blocks within .vue files.

> What did you expect to happen?

No warnings to be flagged. In .scss files there is no warning.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
Button.vue
{
	"resource": "/src/components/forms/Button.vue",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "css-semicolonexpected",
	"severity": 8,
	"message": "semi-colon expected",
	"source": "scss",
	"startLineNumber": 7,
	"startColumn": 17,
	"endLineNumber": 7,
	"endColumn": 26
}
```

**Comment by jeddy3** [comment]
@geert-plessers Thanks for the report and for using the template.

`scss(css-semicolonexpected)` isn't a stylelint warning. I suspect it's coming from another linter, perhaps the one built into VS Code. It's recommended you [turn off](https://github.com/stylelint/vscode-stylelint#optional-but-recommended-setup) other linters when using stylelint.

The `@apply` function is understood by stylelint.

**Comment by geert-plessers** [comment]
@jeddy3 thank you so much. It's not so clear in vscode which linter throws which error. I have all built-in vscode linters disabled so this pointed me towards Vetur, which I also had installed. Apparently this comes with it's own linter for .vue files. Since I don't need this I was able to disable this by placing `"vetur.validation.style": false` in my settings.json file.

**Comment by jeddy3** [comment]
> Apparently this comes with it's own linter for .vue files. Since I don't need this I was able to disable this by placing `"vetur.validation.style": false` in my settings.json file.

Thank you for sharing. That'll be useful information for anyone else who runs into this issue.
